include/opt-sched/Scheduler/aco.h

```diff
@@ -29,22 +31,23 @@ class ACOScheduler : public ConstrainedScheduler {
   virtual ~ACOScheduler();
   FUNC_RESULT FindSchedule(InstSchedule *schedule, SchedRegion *region);
   inline void UpdtRdyLst_(InstCount cycleNum, int slotNum);
-  // Set the initial schedule for ACO 
+  // Set the initial schedule for ACO
   // Default is NULL if none are set.
   void setInitialSched(InstSchedule *Sched);
-  
+
 private:
   pheremone_t &Pheremone(SchedInstruction *from, SchedInstruction *to);
   pheremone_t &Pheremone(InstCount from, InstCount to);
   double Score(SchedInstruction *from, Choice choice);
 
   void PrintPheremone();
 
-  SchedInstruction *SelectInstruction(std::vector<Choice> ready,
-                                      SchedInstruction *lastInst);
+  SchedInstruction *
+  SelectInstruction(const llvm::SmallVectorImpl<Choice> &ready,
+                    SchedInstruction *lastInst);
```

Comment:
I'm not familiar with LLVM best practices. Is `llvm::SmallVectorImpl` a better choice than `llvm::ArrayRef`?

Reply:
http://llvm.org/docs/ProgrammersManual.html says to use "Prefer to use SmallVectorImpl<T> as a parameter type."  It lets you pass SmallVector s of any size.

Reply:
So does ArrayRef. https://lists.llvm.org/pipermail/llvm-dev/2019-May/132127.html

Reply:
I don't think SelectInstruction() s parameter types really matter, It is a private method which is called only once, and is seems unlikely that we would ever want to pass another array type instead of llvm::SmallVector.  I will change it because it seems like you are right as to the standards question.